Sources/SWBBuildSystem/BuildOperation.swift

```diff
@@ -661,8 +661,27 @@ package final class BuildOperation: BuildSystemOperation {
             }
         }
 
+        do {
+            let swiftCacheHits = self.delegate.aggregatedCounters[.swiftCacheHits, default: 0]
+            let swiftCacheMisses = self.delegate.aggregatedCounters[.swiftCacheMisses, default: 0]
+            let clangCacheHits = self.delegate.aggregatedCounters[.clangCacheHits, default: 0]
+            let clangCacheMisses = self.delegate.aggregatedCounters[.clangCacheMisses, default: 0]
+            if swiftCacheHits + swiftCacheMisses > 0 || clangCacheHits + clangCacheMisses > 0 {
+                adaptor.withActivity(ruleInfo: "CompilationCacheMetrics", executionDescription: "Report compilation cache metrics", signature: "compilation_cache_metrics", target: nil, parentActivity: nil) { activity in
+                    if swiftCacheHits + swiftCacheMisses > 0 {
+                        delegate.emit(data: ByteString(encodingAsUTF8: "Swift compiler: \(swiftCacheHits) hits, \(swiftCacheMisses) misses").bytes, for: activity, signature: "compilation_cache_metrics")
+                    }
+                    if clangCacheHits + clangCacheMisses > 0 {
+                        delegate.emit(data: ByteString(encodingAsUTF8: "Clang compiler: \(clangCacheHits) hits, \(clangCacheMisses) misses").bytes, for: activity, signature: "compilation_cache_metrics")
+                    }
```

Comment:
Nit: both of these will print "1 hits" or "1 misses".  It would be nice if we correctly handled the pluralization.